include/rocksdb/db.h

```diff
@@ -353,21 +353,41 @@ class DB {
   // the i'th returned status will have Status::ok() true, and (*values)[i]
   // will store the value associated with keys[i].
   //
-  // (*values) will always be resized to be the same size as (keys).
+  // (*values) is required to be the same size as (keys).
   // Similarly, the number of returned statuses will be the number of keys.
   // Note: keys will not be "de-duplicated". Duplicate keys will return
   // duplicate values in order.
   virtual std::vector<Status> MultiGet(
       const ReadOptions& options,
       const std::vector<ColumnFamilyHandle*>& column_family,
-      const std::vector<Slice>& keys, std::vector<std::string>* values) = 0;
+      const std::vector<Slice>& keys, std::vector<std::string>* values) {
+    assert(values != nullptr);
+    std::vector<PinnableSlice> pinnable_vector(values->size());
+    for (size_t i = 0; i < values->size(); ++i) {
+      assert(!pinnable_vector[i].IsPinned());
+      pinnable_vector[i].ResetSelf(&(*values)[i]);
```

Comment:
Do we absolutely have to add a new API to PinnableSlice? could not we push_back the PinnableSlice(std::string) to the vector one by one?

Reply:
Given that we have disabled the copy and move constructor for PinnableSlice, vector::push_back won't work for PinnableSlice